src/lib/core/media-marshaller/print-hook.service.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Inject, Injectable} from '@angular/core';
+
+import {MediaChange} from '../media-change';
+import {BreakPoint} from '../breakpoints/break-point';
+import {LAYOUT_CONFIG, LayoutConfigOptions} from '../tokens/library-config';
+import {BreakPointRegistry, OptionalBreakPoint} from '../breakpoints/break-point-registry';
+
+/**
+ * Interface to apply PrintHook to call anonymous `target.updateStyles()`
+ */
+export interface HookTarget {
+  activatedBreakpoints: BreakPoint[];
+
+  updateStyles(): void;
+}
+
+@Injectable({providedIn: 'root'})
+export class PrintHookService {
+  protected offlineActivations: BreakPoint[] | null = null;
+
+  constructor(
+      @Inject(LAYOUT_CONFIG) protected layoutConfig: LayoutConfigOptions,
+      protected breakpoints: BreakPointRegistry,
+  ) {
+  }
+
+  /**
+   * Add 'print' mediaQuery: to listen for matchMedia activations
+   */
+  addPrintListener(queries: string[]) {
```

Comment:
This should be called `withPrintListener`, since otherwise it implies that you're adding it to the _service_, and not necessarily to the input array.

Also, add return type